### PR TITLE
feat: implement user contexts

### PR DESCRIPTION
**Pull request by OrKoN**
Spec: https://w3c.github.io/webdriver-bidi/#user-contexts

### CODE REVIEW

File: src/bidiMapper/BidiServer.ts [hunk 1]
```diff
@@ -122,11 +124,31 @@ export class BidiServer extends EventEmitter<BidiServerEvent> {
     parser?: BidiCommandParameterParser,
     logger?: LoggerFn
   ): Promise<BidiServer> {
+    // The default context is not exposed in Target.getBrowserContexts but can
+    // be observed via Target.getTargets. To determine the default browser
+    // context, we check which one is mentioned in Target.getTargets and not in
+    // Target.getBrowserContexts.
```
Comment (sadym-chromium) [inline src/bidiMapper/BidiServer.ts:130]: +1

File: compile_commands.json
Comment (sadym-chromium) [inline compile_commands.json:1]: Artifact?

File: src/bidiMapper/domains/context/BrowsingContextProcessor.ts [hunk 2]
```diff
@@ -106,31 +111,53 @@ export class BrowsingContextProcessor {
...
+
+    try {
+      result = await this.#browserCdpClient.sendCommand('Target.createTarget', {
+        url: 'about:blank',
+        newWindow,
+        browserContextId: userContext === 'default' ? undefined : userContext,
+      });
+    } catch (err) {
+      // https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/devtools/protocol/target_handler.cc;l=1;drc=e80392ac11e48a691f4309964cab83a3a59e01c8
+      if (
```
Comment (sadym-chromium) [inline src/bidiMapper/domains/context/BrowsingContextProcessor.ts:149]: consider extracting into a method

File: src/bidiMapper/domains/context/BrowsingContextProcessor.ts [hunk 3]
```diff
@@ -106,31 +111,53 @@ export class BrowsingContextProcessor {
...
+        (err as Error).message.startsWith(
+          'Failed to find browser context with id'
+        )
+      ) {
+        throw new NoSuchUserContextException(
+          `The context ${userContext} was not found`
+        );
+      }
+      throw err;
```
Comment (sadym-chromium) [inline src/bidiMapper/domains/context/BrowsingContextProcessor.ts:158]: consider extracting into a method

File: tests/browser/test_create_user_context.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+    result = await execute_command(websocket, {
+        "method": "browser.getUserContexts",
+        "params": {}
+    })
+    assert result['userContexts'] == [{
+        'userContext': 'default'
+    }, {
+        'userContext': user_context_id
+    }]
```
Comment (sadym-chromium) [inline tests/browser/test_create_user_context.py:48]: consider sorting, as the order is not important

File: compile_commands.json
Comment (OrKoN) [reply compile_commands.json:1]: Ugh, not even sure where it came from

**Review by sadym-chromium — APPROVED** [review]
LGTM with a minor comments

File: src/bidiMapper/domains/context/BrowsingContextProcessor.ts (hunk 3 above)
Comment (OrKoN) [reply src/bidiMapper/domains/context/BrowsingContextProcessor.ts:158]: only the error handling part of the entire create target part?

**Pull request merged**